sklearn/preprocessing/data.py

```diff
@@ -1698,14 +1698,11 @@ class OneHotEncoder(BaseEstimator, TransformerMixin):
 
     Parameters
     ----------
-    n_values : 'auto', int or array of ints
-        Number of values per feature.
+    classes : 'auto', 2D array of ints or strings or both.
```

Comment:
I don't think 'classes' is a good name. 'values' or 'labels' or 'categories' would be better. I've also not looked into how you're using this purported 2d array, but I don't think a 2d array is appropriate as it constrains all features to have the same number of categories.

I would suggest semantics more compatible with the former implementation, too.

I propose:
````
    values : 'auto', int, list of ints, or list of lists of objects
        - 'auto' : determine set of values from training data.
        - int : values are in ``range(values)`` for all features
        - list of ints : values for feature ``i`` are in ``range(values[i])``
        - list of lists : values for feature ``i`` are in ``values[i]``
````